packages/evm/src/eof/container.ts

```diff
@@ -0,0 +1,229 @@
+import { KIND_CODE, KIND_DATA, KIND_TYPE, MAGIC, TERMINATOR, VERSION } from './constants'
+import { EOFError, validationError } from './errors'
+
+class StreamReader {
+  private data: Buffer
+  private ptr: number
```

Comment:
Can we take the longer more expressive name here (should this mean `pointer` 🤔)?